numpy/core/records.py

```diff
@@ -911,7 +910,7 @@ def fromfile(fd, dtype=None, shape=None, offset=0, formats=None,
     elif isinstance(shape, int):
         shape = (shape,)
 
-    if isfileobj(fd):
+    if hasattr(fd, 'readinto'):
```

Comment:
If it turns out that `isfileobj` can't be changed (see [this discussion](https://github.com/numpy/numpy/pull/16675#discussion_r446736269)), then I suppose replacing these checks is fine. In that case it might be worth adding a comment to explain why `hasattr` is used and which types of objects match this condition.

Reply:
Fixed (added a comment explaining)

Although I do think that we can aim for consistency between this method and `isfileobj`, that is either:
- both of them should be their own functions (Ex. `isfileobj` and `isfilelikeobj`, can try to come up with a better name for the new one as mentioned [here](https://github.com/numpy/numpy/pull/16675#discussion_r446292087)) that way the tests will document which type of objects match. 
- Or both of them should be used directly at the position of the caller, since anyways `isfileobj` is not widely used (only used at `lib/format`)

What do you think?

Reply:
TBH I'm not sure what the right move is. Here are some thoughts I have:
 - I'm against adding *more* `is*obj` functions to `numpy.compat` as I think it has the potential to cause a lot more confusion in the future, especially in the case of `isfileobj` and `isfilelikeobj` (imagine revisiting numpy.compat in the future without any of this discussion as context and trying to figure out why there are these two separate but very similar functions).
 - If `isfileobj` is only used internally in `lib.format`, then maybe removing the instances from the codebase is worthwhile, especially since the function itself is a one-liner. Based on the module docstring for `numpy.compat.py3k`, it seems that these functions were originally intended to bridge the gap between Python 2 and Python 3. Since we no longer support Python 2, it's possible that the `isfileobj` use in `lib.format` is no longer necessary. I'm not very familiar with the differences between Python 2/3 w.r.t. file handling. Maybe you (or someone else) have some insight here?
 - Expanding on the previous point - if it turns out that `isfileobj` is really only useful in the context of Python 2/3 compatibility (and not Python 3.X/3.Y compatiblity), then maybe it would nice to remove it's use internally and deprecate it?

Reply:
Doing a grep on the entire repo I can say that `isfileobj` is only used in `lib.format`. However even I am not aware about the differences of file handling b/w Python 2/3. So I don't know if the impl. of `isfileobj` can be simplified further, since Python 2 is no longer supported.

Also, my thought of being consistent between `isfileobj` and this new one liner was a mere suggestion. Incase we dont get any feedback from more well-versed people about `isfileobj` in a few days, then can we still go ahead and finalize this PR as it is?

Reply:
@rossbar the isfileobj deprecation can be moved into separate issue (not sure about it either)? I think we can move forward with this PR as is.

Reply:
Yes, thanks for the ping @anirudh2290  - I agree that it shouldn't be a blocker. My vote would be to move forward with this PR avoiding functions from `numpy.compat`. I will try to take another look soon.